### PR TITLE
Add `updatePreviousGasParams` to `TransactionController`

### PR DESCRIPTION
**Pull request by OGPoyraz**
## Explanation

This PR adds `updatePreviousGasParams` public method which is used in the extension UI.

## References

Fixes the final part of https://github.com/MetaMask/MetaMask-planning/issues/962

### `@metamask/transaction-controller`

- **Added**: Add `updatePreviousGasParams` public method to `TransactionController`

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [X] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [X] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -1096,6 +1096,62 @@ export class TransactionController extends BaseController<
...
+  ): TransactionMeta {
+    const transactionMeta = this.getTransaction(transactionId);
+
+    if (!transactionMeta) {
+      throw new Error(
+        `Cannot update transaction as no transaction metadata found`,
+      );
+    }
+
+    validateIfTransactionUnapproved(transactionMeta, 'updatePreviousGasParams');
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1129]: Not needed now, but the format of all these pending update methods is so similar, including the validations and `pickBy` and update itself, I wonder if we could remove alot of duplication with a private `updateUnapprovedTransaction`. Perhaps a future refactor task.

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -1096,6 +1096,62 @@ export class TransactionController extends BaseController<
...
+    transactionPreviousGas.previousGas = pickBy(
+      transactionPreviousGas.previousGas,
+    );
+
+    // merge updated previous gas values with existing transaction meta
+    const updatedMeta = merge(transactionMeta, transactionPreviousGas);
+
+    this.updateTransaction(
+      updatedMeta,
+      'TransactionController:updatePreviousGasParams - gas values updated',
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1149]:
```suggestion
      'TransactionController:updatePreviousGasParams - Previous gas values updated',
```

File: packages/transaction-controller/src/types.ts [hunk 3]
```diff
@@ -106,6 +106,26 @@ type TransactionMetaBase = {
    */
   originalGasEstimate?: string;
 
+  /**
+   * The original gas estimation before updating the gas values, set by UI.
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/types.ts:110]:
```suggestion
   * The previous gas properties before they were updated.
```

File: packages/transaction-controller/src/TransactionController.ts (hunk 1 above)
Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts:1129]: Exactly Matthew, and those update functions located in the scoped files (e.g `updateGasFees`, `updateGas`, `updateTransactionHistory`) only manipulating the passed transaction metadata, having the same naming convention with public update methods (`updateX`). It's a minor one but would be much better if we have clear separation somehow on their name imo.

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**